app/data_client.go

```diff
@@ -109,6 +109,37 @@ type TabularData struct {
 	TimeReceived  time.Time
 }
 
+// ExportTabularDataReturn represents the result of an ExportTabularData API call.
+type ExportTabularDataReturn struct {
+	OrganizationID   string
+	LocationID       string
+	RobotID          string
+	RobotName        string
+	PartID           string
+	PartName         string
+	ComponentName    string
+	ComponentType    string
+	MethodName       string
+	TimeCaptured     time.Time
+	MethodParameters map[string]any
+	Tags             []string
+	Payload          map[string]any
+}
+
+// ExportTabularDataStream is a stream that returns ExportTabularDataReturns.
+type ExportTabularDataStream struct {
+	Stream pb.DataService_ExportTabularDataClient
+}
+
+// Next gets the next ExportTabularDataReturn.
+func (e *ExportTabularDataStream) Next() (ExportTabularDataReturn, error) {
+	streamResp, err := e.Stream.Recv()
+	if err != nil {
+		return ExportTabularDataReturn{}, err
```

Comment:
This can return a pointer to a `ExportTabularDataReturn` so then we can write `return nil, err` here.